mmv1/products/servicedirectory/api.yaml

```diff
@@ -38,10 +38,12 @@ objects:
       api: 'https://cloud.google.com/service-directory/docs/reference/rest/v1beta1/projects.locations.namespaces'
     iam_policy: !ruby/object:Api::Resource::IamPolicy
       exclude: false
-      parent_resource_attribute: 'name'
+      parent_resource_attribute: 'namespace_id'
       method_name_separator: ':'
       fetch_iam_policy_verb: :POST
       set_iam_policy_verb: :POST
+      base_url: "projects/{{project}}/locations/{{location}}/namespaces/{{namespace_id}}"
+      import_format: ["projects/{{project}}/locations/{{location}}/namespaces/{{namespace_id}}"]
```

Comment:
This is a breaking change 💔 

The error that happens with the old code is that there's a 404 when trying to get the IAM policy on the Namespace:

> POST /v1beta1/tf-test-example-namespace6agdt0mwhx:getIamPolicy?alt=json 

This appears to be due to [the `qualifyNamespaceUrl` function](https://github.com/hashicorp/terraform-provider-google-beta/blob/b1b84857f47513621fb09012453e010c5ef73fce/google-beta/iam_service_directory_namespace.go#L148-L155) producing the wrong URL that lacks project/location etc info. I tried updating the test to have a long-form name so the URL is complete but it didn't work for me and I figured it was worth getting some advice!

Reply:
yeah, this is definitely a breaking change for users of this resource. Basically to justify this change in a minor release we need to be reasonably sure that if someone was using a configuration with `name` it just wouldn't work for them - is that the case here? Like, can these IAM resources be used to _set_ the IAM policy but then not get or update it, or do they just not work at all?

Reply:
Just to double-check, does setting base_url & import_format to just contain `{{name}}` work?

Reply:
> Just to double-check, does setting base_url & import_format to just contain {{name}} work?

When this is set for the Namespace resource and I run the `TestAccServiceDirectoryNamespaceIamMemberGenerated` test it fails on the first step (`test_step_number=1
` mentioned in the copy/paste output below) because it's making a POST request to  `/v1beta1/projects:getIamPolicy?alt=json`  instead of `/v1beta1/projects/XXX/locations/XXX/namespaces/tf-test-xxx:getIamPolicy?alt=json`

```
2023-01-27T17:28:41.786Z [WARN]  sdk.helper_resource: Error running Terraform CLI command: test_step_number=1
  error=
  | exit status 1
  | 
  | Error: Error retrieving IAM policy for servicedirectory namespace "projects": googleapi: got HTTP response code 404 with body: <!DOCTYPE html>
  | <html lang=en>
  |   <meta charset=utf-8>
  |   <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  |   <title>Error 404 (Not Found)!!1</title>
  |   <style>
  |     *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  |   </style>
  |   <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  |   <p><b>404.</b> <ins>That’s an error.</ins>
  |   <p>The requested URL <code>/v1beta1/projects:getIamPolicy?alt=json</code> was not found on this server.  <ins>That’s all we know.</ins>
  | 
  | 
  |   with google_service_directory_namespace_iam_member.foo,
  |   on terraform_plugin_test.tf line 14, in resource "google_service_directory_namespace_iam_member" "foo":
  |   14: resource "google_service_directory_namespace_iam_member" "foo" {
```

Reply:
> Basically to justify this change in a minor release we need to be reasonably sure that if someone was using a configuration with name it just wouldn't work for them - is that the case here? Like, can these IAM resources be used to set the IAM policy but then not get or update it, or do they just not work at all?

I manually tested creating a `google_service_directory_namespace` and a  `google_service_directory_namespace_iam_binding` resource and can create and update the iam_binding without any errors. I'm not sure why there isn't a 404 like in the tests 😕 

**Something that doesn't seem to work is import of `google_service_directory_namespace_iam_binding`**:

```hcl
resource "google_service_directory_namespace" "example" {
  provider     = google-beta
  namespace_id = "tf-test-example-namespace-sarahfrench"
  location     = "us-central1"

  labels = {
    key = "value"
    foo = "bar"
  }
}

resource "google_service_directory_namespace_iam_binding" "foo" {
  provider = google-beta
  name     = google_service_directory_namespace.example.name
  role     = "roles/viewer"
  members  = ["user:admin@hashicorptest.com"]
}
```

After applying the plan to create these I then remove it from state and attempt to import the binding [following the provider docs](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/service_directory_namespace_iam#import) and it fails:

```bash
terraform state rm google_service_directory_namespace_iam_binding.foo
# works

terraform import google_service_directory_namespace_iam_binding.foo "projects/[PROJECT_ID]/locations/us-central1/namespaces/tf-test-example-namespace-sarahfrench roles/viewer"

# Error: Import id "" doesn't match any of the accepted formats: [projects/(?P<project>[^/]+)/locations/(?P<location>[^/]+)/namespaces/(?P<namespace_id>[^/]+) (?P<project>[^/]+)/(?P<location>[^/]+)/(?P<namespace_id>[^/]+) (?P<location>[^/]+)/(?P<namespace_id>[^/]+)]
```